### PR TITLE
Do not check for an existing package map if a command is going to run…

### PR DESCRIPTION
**Pull request by jason-simmons**
… "pub get" to update the package map

Also remove a redundant check for the --pub option in the run command

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
@devoncarew is the right reviewer here.  Do we have an easy way to test this kind of thing? I suspect not yet. :/  Thanks for fixing!

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -157,10 +157,12 @@ abstract class FlutterCommand extends Command {
       }
     }
 
-    String error = PackageMap.instance.checkValid();
-    if (error != null) {
-      printError(error);
-      return false;
+    if (!(_usesPubOption && argResults['pub'])) {
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: Perhaps add a comment that we only validate the pubspec here if we're not going to run `pub` later on?

**Comment by devoncarew** [comment]
lgtm; I think the check in `run()` existed before we added code to run pub for certain commands.

File: packages/flutter_tools/lib/src/runner/flutter_command.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: done

**Pull request merged**